src/components/deploymentConfig/DeploymentTemplateView/DeploymentTemplateEditorView.tsx

```diff
@@ -170,81 +237,96 @@ export default function DeploymentTemplateEditorView({
         }
     }
 
-    const renderCodeEditor = (): JSX.Element => {
-        return (
-            <div
-                className={`form__row--code-editor-container dc__border-top-n1 dc__border-bottom-imp ${
-                    isDeleteDraftState && !state.showReadme ? 'delete-override-state' : ''
-                }`}
+    const selectedOptionId = state.selectedCompareOption?.id
+    const isIdMatch = selectedOptionId === -1 || selectedOptionId === Number(envId)
+
+    const valueLHS = isValues
+        ? (isIdMatch ? defaultValue : state.fetchedValues[selectedOptionId]) || ''
+        : (isIdMatch ? defaultValue : state.fetchedValuesManifest[selectedOptionId]) || ''
+
+    const renderCodeEditor = (): JSX.Element => (
```

Comment:
Please break this method in smaller component

Reply:
done